src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -152,14 +143,31 @@ public function validate($entity, Constraint $constraint)
             reset($result);
         }
 
-        /* If no entity matched the query criteria or a single entity matched,
-         * which is the same as the entity being validated, the criteria is
-         * unique.
-         */
-        if (0 === count($result) || (1 === count($result) && $entity === ($result instanceof \Iterator ? $result->current() : current($result)))) {
+        if (0 === count($result)) {
             return;
         }
 
+        if (1 === count($result)) {
+            $match = $result instanceof \Iterator ? $result->current() : current($result);
+
+            if ($object === $match) {
+                return;
+            }
+
+            $method = $constraint->isEntityToUpdateMethod;
```

Comment:
We must add support to `\Closure` and `array` callable too.

Reply:
This method is just for saying an entity is the one we're updating or not I don't think we need more complicated?

Reply:
This constraint can be used in PHP flat and maybe for advanced checking where I need some dependencies as service for instance.

Reply:
It should be a `callable` yes. I would not like to add this method to my dto. Defeats decoupling IMHO.

We can do the standard approach no? Check for callable, then for method on current context which we do in validator component i believe.